navigator/backend/src/test/scala/com/digitalasset/navigator/backend/DamlConstants.scala

```diff
@@ -308,6 +306,17 @@ case object DamlConstants {
     )
   )
 
+  def complexGenMapT(keyTyp: DamlLfIface.Type, valueTyp: DamlLfIface.Type) =
+    DamlLfTypePrim(DamlLfPrimType.GenMap, DamlLfImmArraySeq(keyTyp, valueTyp))
+
+  val complexGenMapV = V.ValueGenMap(
+    ImmArray(
+      treeV -> V.ValueInt64(1),
+      treeLeftV -> V.ValueInt64(2),
+      treeLeftV -> V.ValueInt64(3)
```

Comment:
Can you have duplicate keys in genmap?

Reply:
The ledger will never return duplicate key, but you can submit with duplicate key (the last one will override the formers)